verify-node/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+extern crate wasm_bindgen;
+
+use ed25519_dalek::Signature;
+use holo_config_core::AdminPublicKey;
+use std::str::FromStr;
+use wasm_bindgen::prelude::*;
+
+// Use `wee_alloc` as the global allocator.
+#[global_allocator]
+static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+pub type JsResult<T> = Result<T, JsValue>;
+
+macro_rules! jserr {
```

Comment:
This would be better kept as a function:

```
fn into_js_error(err: impl std::error::Error) -> JsValue {
   js_sys::Error::new(&err.to_string()).into()
}
```

For example, see: https://blog.cloudflare.com/building-fast-interpreters-in-rust/

Reply:
Agreed.  This was imported from some other code.  The  original function defined the `Fail` trait, which was not satisfied by some errors, and I couldn't figure out how to address it, so I revereted back to this working method.